doc/tutorial/tutorial.mdk

```diff
@@ -1255,6 +1297,13 @@ it is:
 In the `Cons` case of `rev_involutive` we are explicitly applying not just
 the induction hypothesis but also the `snoc_cons` lemma we proved above.
 
+~KK
+At this point it would be nice to have an explaination on how these two
+functions actually help F\* (or Z3) prove the property. It took me quite a while
+to figure out how this might work and I'm still not entirely certain. Again,
+this depends on who the audience of this tutorial is.
+~
+

```

Comment:
'Rolling out' fully for something simple would be nice. I struggle with this just as much as you.